Physical test for the 2D steady state heat equation

twh5306 opened the pull request:
2d heat equation which is compared to the analytical solution for error calcuation.  Note the the ElementL2Error and NodalL2Error are oders of magnitude in difference, but both show very small error.

Closes #6181

Commit:
for issue number 6181 this is a code shows a 2d heat equation which is compared to the analytical solution for error calcuation.  Note the the ElementL2Error and NodalL2Error are oders of magnitude in difference, but both show very small error

physical test for the 2D heat equaiton  closes #6196

-m for issue number 6181 this is a code shows a 2d heat equation which is compared to the analytical solution for error calcuation.  Note the the ElementL2Error and NodalL2Error are oders of magnitude in difference, but both show very small error

-m sorry for all the commits

Commit:
for issue number 6181 this is a code shows a 2d heat equation which is compared to the analytical solution for error calcuation.  Note the the ElementL2Error and NodalL2Error are oders of magnitude in difference, but both show very small error

physical test for the 2D heat equaiton  closes #6196

Commit:
-m for issue number 6181 this is a code shows a 2d heat equation which is compared to the analytical solution for error calcuation.  Note the the ElementL2Error and NodalL2Error are oders of magnitude in difference, but both show very small error

jwpeterson commented:
I think we have a number of tests of the method of manufactured solutions (MMS) for the heat equation already in the test suite... so I'm curious what you think is different about this one?  Also 150x150 is a fairly large problem for the regression test suite.

dschwen commented:
@twh5306 could you clean up the PR title to make understandable ;-). Which issue number does this reference? That number goes in the PR description, not the title.

jwpeterson commented:
One other comment, using `3.14159265` for the value of pi is not advisable... you won't get proper rates of convergence on very fine grids, etc.

tonkmr commented:
@jwpeterson, the plan is to add several simulations that have known close form analytical solutions for verification purposes. This is different than comparing against a manufactured solution because it is an actual physical solution.

jwpeterson commented:
> This is different than comparing against a manufactured solution because it is an actual physical solution.

I guess I just don't see the difference between "physical" and "manufactured" as far as code verification is concerned.

friedmud commented:
@jwpeterson actually, we don't have that many good tests for heat conduction and this one looks like a good addition.

A couple of things:

1.  Put the issue number into the main text of the PR (remove it from the title)
2.  Need to put a # sign in front of the issue number in the commit message
3.  Use `pi` instead of 3.1415926...

Thanks!

friedmud commented:
Oh - I agree that 150x150 is a bit excessive.  Maybe 30x30 or so?

jwpeterson commented:
> @jwpeterson actually, we don't have that many good tests for heat conduction and this one looks like a good addition.

We already have
```
./tests/postprocessors/mms_sine/2_d_mms_sine_postprocessor_test.i
./tests/postprocessors/mms_sine/3_d_mms_sine_postprocessor_test.i
```
and they are more general since they are capable of testing convection-diffusion-reaction.  I don't mind adding a new test as long as it tests something besides a different form of the exact solution...

friedmud commented:
But those aren't actually testing HeatConduction and don't use material properties at all.  Yes, the equations that are solved are similar... but the code that is tested is not.

Plus, this has the bonus of being a nice, simple example for people to see how to use HeatConduction Kernels.

Finally, I'll say that this is a new developer that is obviously just getting started.  I appreciate their willingness to contribute something useful so I wouldn't reject it.

aeslaughter commented:
I am with @friedmud on this.

jwpeterson commented:
Well, I would at least like to see a convergence plot under uniform refinement that confirms we actually get the right order. This does not need to be part of the test, just attach an image to the PR.

twh5306 commented:
@jwpeterson  using the parsed function, can i reference pi without writing out a numerical value?

jwpeterson commented:
Yes, you should be able to just use "pi".  If that doesn't work, please let @dschwen know.

twh5306 commented:
will do

twh5306 commented:
@jwpeterson  when i tried to use pi instead of 3.14159265 i got an error, Unable to find function 10/(sinh(pi))*sin(pi*x*0.5)*sinh(pi*y*0.5)" and everything runs fine with the approximation

twh5306 commented:
@jwpeterson I tried pi instead of 3.14159265 and got the following error "Unable to find function 10/(sinh(pi))*sin(pi*x*0.5)*sinh(pi*y*0.5)"

twh5306 commented:
the times symbols didn't show in the comment, but they are there

andrsd commented:
> the times symbols didn't show in the comment, but they are there

I edited your comment so it shows up correctly.

jwpeterson commented:
> I tried pi instead of 3.14159265 and got the following error

I think what you want to do is have a `[Functions]` block in your input file, e.g.
```
[Functions]
  [./myfunc]
    type = ParsedFunction
    value = sin(pi*x)*sin(pi*y)
  [../]
[]
```
and then in the `FunctionDirichletBC` you would say 

```
function = myfunc
```

and it should work.

aeslaughter committed:
Removes Postprocessor/VectorPostprocessor Warehouses

(refs #5680)

aeslaughter committed:
Populate list of output names prior to construction for UserObjects 'outputs' parameter works

(refs #5680)

dschwen committed:
Stray semicolol warning (#1777)

permcody committed:
Merge pull request #6230 from dschwen/telwarn_1777

Fix stray semicololon warning

twh5306 commented:
@jwpeterson I made those functional adjustments and it appears to be working.  Are there any other issues you see that I should change before I re-submit? (I'm working on the plot you requested now)

dschwen commented:
There is no need to "resubmit", just push your changes to the same branch. Please check the pre-test results, which is failing.

jwpeterson commented:
> . Are there any other issues you see

Did you make the grid smaller?  30x30 is probably at the top end of what we want in the regression test suite, although it would be good if you could time it yourself to be sure.  Anything more than about 5 seconds in opt mode is probably too slow.

twh5306 commented:
yes I made the grid 30x30, and it runs pretty fast

twh5306 commented:
![error_pic_for_final_2d_steady](https://cloud.githubusercontent.com/assets/15001683/12491417/f030d762-c048-11e5-9f63-2e7c9b6e98f3.png)

This shows increased accuracy with a finer mesh as expected, but the two different error calculators in MOOSE are 1 to 2 orders of magnitude off from each other

jwpeterson commented:
What is "number of partitions"?

The purpose here is not to show that the error is decreasing, although that is good, it is to confirm the rate at which it decreases.  The easiest way to do this is by plotting the log of the error vs. log(h), where h is the length of a side of an element.  Make the plot for 10x10, 20x20, 40x40, 80x80 element meshes (hence h will be 2/10, 2/20, 2/40, ... in your case).  This plot should look like a straight line, and we are interested in the slope of that line.  Also, we are more interested in the ElementL2Error, for linear problems like this the NodalL2Error is typically super-convergent

dschwen committed:
Add MaterialUserObject class (#6233)

jwpeterson committed:
Merge pull request #6232 from aeslaughter/control-parallel-6231

Re-design control tests to the DiracKernel point is not on element edge

Commit:
Merge branch 'devel' of https://github.com/twh5306/moose into devel
I just want to this to finally merge

Commit:
sorry for all the commits

dschwen committed:
Add interface for MaterialUserObjects (#6233)

dschwen committed:
Use MaterialUserObject in HEVP system (#6233)

WilkAndy committed:
Rachel and Andy work on improving tensor_mechanics plasticity performance

TensorMechanicsPlasticModel now has returnMap and consistentTangentOperator methods
that any derived class can override.  An example is TensorMechanicsPlasticJ2

ComputeMultiPlasticityStress now uses these custom returnMap and consistentTangentOperator
methods if possible to dramatically speed up the return-map process.  If the custom
versions fail, or are not coded, or if more than one model is plastically active,
the return-map algorithm falls back to the previous algorithm.

MultiPlasticityRawComponentAssembler has a returnMapAll function that tries all
the custom returnMap functions.

Fixes #5667

WilkAndy committed:
Tests of Rachel and Andy work.

Most of these are just re-goldings due to zero iterations now being required.
The j2_plasticity_vs_LSH stuff now does a CSVDiff instead, and runs for end_time=0.5

Fixes #5667

jwpeterson committed:
Fix unused variable warning in FEProblem.

Refs #1777.

jwpeterson committed:
Delete tabs and fix unused variable warnings in ICE updater.

Refs #1777.

jwpeterson committed:
Fix warnings and spacing around operators in InputParameters.

Refs #1777.

Commit:
Add KKS ternary example input file Refs #6244

permcody committed:
Stop calling Threads::pthread_unique_id()

This PR is the first of several threading fixups in the framework.
This first avoids calling the broken pthread_unique_id() method
in libMesh and hands out IDs based on a properly protected queue
instead.

refs #6245

Commit:
Fix typos and Petsc options in existing KKS examples so they can run in parallel. Refs #6244

dschwen committed:
Make c coupling optional and default to variable (#6239)

WilkAndy committed:
solid_mechanics tests changed

cracking_exponential.exodiff now has floor set for off-diagonal stress components

crack_2d_sent.i now uses tighter convergence tolerances which means a re-golding of crack_2d_sent_out.e was necessary.

Fixes #6243

sapitts committed:
Tightened solver tolerances and set abs_zero in 2D planestrain cases Refs #6119

dschwen committed:
Fix and simplify ACInterface residual (#6194)

dschwen committed:
Allow pi etc. in auto-parsed functions (#6250)

dschwen committed:
Remove wrong auto-parse parameter (#6250)

dschwen committed:
Fix tests to compute required second derivative (#6194)

dschwen committed:
Suppress execute_on parameter and set to custom (#6233)

dschwen committed:
Fix off-diagonal Jacobian for ACinterface (#6194)

permcody committed:
Merge pull request #6238 from jwpeterson/fix_warnings

Fix warnings in ICEUpdater, framework

permcody committed:
Merge pull request #6248 from dschwen/simplifysoret_6239

Make c coupling optional and default to variable

permcody committed:
Merge pull request #6249 from WilkAndy/solid_mech_tests_6243

solid_mechanics tests changed

permcody committed:
Merge pull request #6247 from laagesen/kks_examples_6244

Update KKS phase-field model examples

permcody commented on modules/heat_conduction/tests/verify_against_analytical/2d_steady_state_final_prob.i:
```diff
@@ -0,0 +1,119 @@
+
+# Note that the thermal conductivity, specific heat, and density in this problem
+# Are set to 1, and need to be changed to the constants of the material being
+# Analyzed
```
Since you have an analytical solution for this problem, I'd like you to go ahead and describe the problem setup and show the solution here in the test file header. If you have a reference to where we can quickly find the solution that would be very valuable as well. If you want to take a look at some well documented tests. Look at the `.i` files in this directory: https://github.com/idaholab/moose/tree/devel/modules/combined/tests/gap_heat_transfer_htonly

permcody commented:
I'd like to get this test merged soon. If you'll address my last comment, we'll get this in. Also, it'd be nice if you just squashed your commits down to a single commit before pushing. You just do an interactive rebase `git rebase -i HEAD~<n>` where `<n>` is the number of commits in your branch. Then you mark all of the commits except the very top one as `s` instead of `pick`. I'm happy to help out if you have any questions.

andrsd committed:
AuxKernel provides _current_node from the correct mesh

Current node was coming from _var, which was always the non-displaced
version.  It is now coming from Assembly, which respects the displaced
non-displaced variant.

Closes #6086

jwpeterson committed:
Merge pull request #6246 from permcody/thread_6245

Stop calling Threads::pthread_unique_id()

twh5306 commented:
Ok @permcody I'll make the changes before the end of the day tomorrow and try to squash them.  if i have any questions on how to do that I'll make sure to ask here, thanks!

jwpeterson committed:
Merge pull request #6251 from dschwen/autoparse_6250

Allow pi etc. in auto-parsed functions

jwpeterson committed:
Merge pull request #6255 from andrsd/current-node-elem-aux

AuxKernel provides _current_node from the correct mesh

jwpeterson committed:
Merge pull request #6252 from sapitts/correctPlStrainTol_6119

Tightened solver tolerances and set abs_zero in 2D planestrain cases …

permcody committed:
Merge pull request #6235 from dschwen/threaduo_6233

MaterialUserObject - on-demand recalculatable "material properties"

dschwen committed:
Address comments (#6194)

dschwen committed:
Remove unused derivatives (#6194)

dschwen committed:
Make stuff const (#6194)

WilkAndy committed:
Address dschwen comments:

 - made called_from an enum
 - created _dummy_pm and _cumulative_pm which are member variables to avoid too much re-assigning stuff
 - made trial_stress_inadmissible a bool
 - made debug_fspb an enum instead of an int, which required heaps of one-line changes in the tests

Also fix a bug in MultiPlasticityRawComponentAssembler::returnMapAll

Fixes #5667

dschwen committed:
Prevent queue declaration for OpenMP (#6257)

permcody committed:
Merge pull request #6260 from dschwen/openmpfix_6257

ParallelUniqueID compile fix

permcody committed:
Merge pull request #6253 from dschwen/coupledacinterface_6194

Fix ACInterface residuals and Jacobians yet again

dschwen committed:
Thus ends the short life of getMaterialUserObject (#6233)

dschwen committed:
Apply queue fix to .C file (#6257)

dschwen committed:
Roll back to getUserObjectByName (#6233)

permcody committed:
Merge pull request #6262 from dschwen/puidrefix_6257

Apply queue fix to .C file

dschwen committed:
Revise clobebr target (#6258)

dschwen committed:
Rename to LocalElementUserObject (#6233)

dschwen committed:
Rename to isLocalUserObject (#6233)

dschwen committed:
Rename Local to Discrete per @friedmud (#6233)

friedmud committed:
Merge pull request #6263 from dschwen/threaduo_6233

Make getUserObject* return the correct thread copy for MaterialUserObjects

jwpeterson committed:
Merge pull request #6264 from dschwen/clobber_6258

Revise clobber target

dschwen committed:
Allow use of RandomInterface in const objects (#6267)

permcody committed:
Merge pull request #6268 from dschwen/rngmutable_6267

Allow use of RandomInterface in const objects

twh5306 commented:
I'm sorry for the delay, but I got sick at the end of last week.  I keep getting the following error

error: could not apply e11ad50... -m for issue number 6181 this is a code shows a 2d heat equation which is compared to the analytical solution for error calcuation.  Note the the ElementL2Error and NodalL2Error are oders of magnitude in difference, but both show very small error

-m and on was just part of the original commit

aeslaughter committed:
Adds --minimal for creating a MooseApp without an input file

(refs #6126)

aeslaughter committed:
Adds test for running with --minimal mode

(refs #6126)

aeslaughter committed:
Adds unit test for creating a MooseApp without an input file

(closes #6126)

permcody commented:
Can you give us the exact error message? It sounds like you may have a
conflict but I don't see why. I don't believe you touched any other files
that would have been modified in-between. It's true that Daniel fixed up
your commit message in the current PR but I doubt you pulled his version
back down. Show us the commands you are typing and the errors and we'll
give you a hand in getting them fixed up.

Cody

On Tue, Feb 2, 2016 at 2:14 PM twh5306 <notifications@github.com> wrote:

> I'm sorry for the delay, but I got sick at the end of last week. I keep
> getting the following error
>
> error: could not apply e11ad50
> <https://github.com/idaholab/moose/commit/e11ad5035a691c2840134ea8ad7ba6ef4a6bb2b4>...
> -m for issue number 6181 this is a code shows a 2d heat equation which is
> compared to the analytical solution for error calcuation. Note the the
> ElementL2Error and NodalL2Error are oders of magnitude in difference, but
> both show very small error
>
> -m and on was just part of the original commit
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6196#issuecomment-178823853>.
>

permcody committed:
Merge pull request #6234 from aeslaughter/pp-wh-5680-v2

Remove Postprocessor/VectorPostprocessor Warehouses

twh5306 commented:
this screenshot shows my commands and the error.  after the i HEAD I deleted the pick in all but the first statement, replaced them with s, then hit esc followed by :wq
![screen shot 2016-02-03 at 1 13 07 pm](https://cloud.githubusercontent.com/assets/15001683/12792112/46a01110-ca78-11e5-8e3e-ca67146bef7f.png)

permcody commented:
Ok so there is a conflict between your work and the work that has been
pushed in since there. You need to type "git diff" and/or "git status".
That'll show you which files are in conflict. You have to edit them to
resolve the conflicts. "git" guides you and tells you what commands to
type. After you fix the problem you type "git add" on the affected files.
Finally you type git rebase --continue.



On Wed, Feb 3, 2016 at 11:16 AM twh5306 <notifications@github.com> wrote:

> this screenshot shows my commands and the error. after the i HEAD I
> deleted the pick in all but the first statement, replaced them with s, then
> hit esc followed by :wq
> [image: screen shot 2016-02-03 at 1 13 07 pm]
> <https://cloud.githubusercontent.com/assets/15001683/12792112/46a01110-ca78-11e5-8e3e-ca67146bef7f.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6196#issuecomment-179386677>.
>

jwpeterson committed:
Merge pull request #6271 from aeslaughter/minimal-app-6126

Adds --minimal option for running MooseApp w/o an input file.

jwpeterson committed:
Merge pull request #6237 from WilkAndy/returnMappingTM_5667

Return mapping optimization in tensor_mechanics

dschwen committed:
Move templated member implementation into header (#6275)

aeslaughter committed:
Revert "Revise clobebr target (#6258)"

This reverts commit 5dbb349f7eb24434c8cba82b1eed246f50fe0df9.

The new "clobber" command wipes out the .git directory, which is bad.

jwpeterson committed:
Merge pull request #6277 from aeslaughter/clobber

Revert "Revise clobebr target (#6258)"

permcody committed:
Merge pull request #6276 from dschwen/exprbtemplate_6275

Move templated member implementation into header

Commit:
Merge branch 'devel' of https://github.com/twh5306/moose into devel

twh5306 commented:
@permcody did that work?

permcody commented:
Unfortunately no, if you look at the commits on this ticket, you'll see
hundreds. There should only be one or two if you've rebased properly.
On Fri, Feb 5, 2016 at 1:33 PM twh5306 <notifications@github.com> wrote:

> @permcody <https://github.com/permcody> did that work?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6196#issuecomment-180545891>.
>

permcody commented:
So what happened @twh5306? Why did you close this?

twh5306 commented:
I'm re-cloning my fork and re-installing in a attempt to start from a blank slate.  I saved the file separately, and i'll open a new PR once I'm done re-cloning and re-installing libmesh
